More dynamicDowncast and RefPtr in WebCore/page/EventHandler

annevk opened the pull request:
#### 06ee12da4dccd521b19af5fc4dd5e6073c78ef05
<pre>
More dynamicDowncast and RefPtr in WebCore/page/EventHandler
<a href="https://bugs.webkit.org/show_bug.cgi?id=266212">https://bugs.webkit.org/show_bug.cgi?id=266212</a>

Reviewed by Chris Dumez.

Also add some CheckedPtr as per review comments.

* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::handleMousePressEvent):
(WebCore::EventHandler::startPanScrolling):
(WebCore::EventHandler::subframeForTargetNode):
(WebCore::isSubmitImage):
(WebCore::EventHandler::enclosingScrollableArea):
(WebCore::contentFrameForNode):
(WebCore::findDropZone):
(WebCore::EventHandler::widgetForEventTarget):
(WebCore::widgetForElement):
(WebCore::EventHandler::handleWheelEventInAppropriateEnclosingBox):
(WebCore::EventHandler::capsLockStateMayHaveChanged const):
(WebCore::EventHandler::passWidgetMouseDownEventToWidget):

Canonical link: <a href="https://commits.webkit.org/271898@main">https://commits.webkit.org/271898@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bffd0c376e5c2ea7fcb0f16d83d235a983dc15a8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/29938 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/8604 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/31256 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/32450 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/27073 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/30598 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/10790 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/5859 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/27122 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/30232 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/7213 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/25542 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/6154 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/6322 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/26631 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/33786 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/27308 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/27075 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/32496 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/6255 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/4431 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/30287 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/7991 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7105 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/6997 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/6781 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/page/EventHandler.cpp:
```diff
@@ -1198,9 +1198,10 @@ void EventHandler::didPanScrollStop()
 void EventHandler::startPanScrolling(RenderElement& renderer)
 {
-    if (!is<RenderBox>(renderer))
+    auto* renderBox = dynamicDowncast<RenderBox>(renderer);
```
Would be nice to use CheckedPtr here.

cdumez commented on Source/WebCore/page/EventHandler.cpp:
```diff
@@ -1395,20 +1396,21 @@ RefPtr<Frame> EventHandler::subframeForTargetNode(Node* node)
 
-    auto renderer = node->renderer();
-    if (!is<RenderWidget>(renderer))
+    auto* renderWidget = dynamicDowncast<RenderWidget>(node->renderer());
```
Would be nice to use CheckedPtr here.

cdumez commented on Source/WebCore/page/EventHandler.cpp:
```diff
@@ -3244,15 +3249,15 @@ bool EventHandler::handleWheelEventInAppropriateEnclosingBox(Node* startNode, co
     // RenderListBox is special because it's a ScrollableArea that the scrolling tree doesn't know about.
-    if (is<RenderListBox>(initialEnclosingBox))
-        handleWheelEventPhaseInScrollableArea(downcast<RenderListBox>(initialEnclosingBox), wheelEvent);
+    if (auto* renderListBox = dynamicDowncast<RenderListBox>(initialEnclosingBox))
```
Would be nice to use CheckedPtr here.

annevk committed:
More dynamicDowncast and RefPtr in WebCore/page/EventHandler
https://bugs.webkit.org/show_bug.cgi?id=266212

Reviewed by Chris Dumez.

Also add some CheckedPtr as per review comments.

* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::handleMousePressEvent):
(WebCore::EventHandler::startPanScrolling):
(WebCore::EventHandler::subframeForTargetNode):
(WebCore::isSubmitImage):
(WebCore::EventHandler::enclosingScrollableArea):
(WebCore::contentFrameForNode):
(WebCore::findDropZone):
(WebCore::EventHandler::widgetForEventTarget):
(WebCore::widgetForElement):
(WebCore::EventHandler::handleWheelEventInAppropriateEnclosingBox):
(WebCore::EventHandler::capsLockStateMayHaveChanged const):
(WebCore::EventHandler::passWidgetMouseDownEventToWidget):

Canonical link: https://commits.webkit.org/271898@main
